src/sb/app/chat/data.cljc

```diff
@@ -126,13 +114,48 @@
 (def chat-fields-full
   (conj chat-fields-meta {:chat/messages message-fields}))
 
+(defn get-chat-id [{:as params :keys [account-id chat-id other-id]}]
+  (or chat-id
+      (-> (db/where [[:chat/participants other-id]
+                     #(some (comp #{(sch/unwrap-id account-id)}
+                                  :entity/id
+                                  :membership/member)
+                            (:chat/participants %))])
+          first
+          sch/wrap-id)))
+
 (q/defquery chat
   "Get a chat by chat-id"
   {:prepare [az/with-account-id!]}
   [{:as params :keys [account-id chat-id]}]
   (ensure-participant! account-id (dl/entity chat-id))
   (q/pull chat-fields-full chat-id))
 
+(q/defquery proto-chat
```

Comment:
is this a thing for dev/testing purposes?

Reply:
No it's actually being used to load the chat participants and entity for a new chat into the client.